docs/porting/libraries.md

```diff
@@ -1,5 +1,153 @@
-> 
-> This topic hasn’t been written yet! You can track the status of this issue through our public GitHub issue tracker. Learn how you can contribute on GitHub.
+blah blah some intro
+
+## Prerequisites
+
+Before you begin, understand that this article assumes that you are using Windows and Visual Studio.
+
+It's recommended that you read the [Porting Overview](overview.md) article before you begin.  It will give you an idea of the journey you're about to embark on.
+
+## Getting your dependencies and .NET Framework APIs in check
+
+The two biggest difficulties in porting libraries to .NET Core is resolving issues with 3rd party dependencies and .NET Framework APIs that aren't compatible with .NET Core.
+
+If you haven't resolved issues with 3rd party dependencies, [please do so before continuing](overview.md).
+
+Next, make sure you [understand the API Portability tool](tooling.md) and can generate portability reports for targeting .NET Core.  You'll have to spend some time analyzing your assemblies and coming up with a plan for what to do with any incompatible types used.
+
+Here's a set of guidelines and questions you may find helpful.
+
+1. Understand where in the code each incompatible type is being used.
+
+   a.  Understand the nature of the incompatible types.  Are they small in number, but used frequently?  Are they large in number, but used infrequently?  Is their use concentrated, or is it spread throughout your code?
+   
+   b. Is it easy to isolate code that's largely incompatible from code that's large compatible?
+   
+   c. Would you need to refactor your code so that you can deal with incompatible types better?
+
+2. If you have assemblies that are mostly incompatible, is it worth leaving them on .NET Framework for now?  Note that this will depend heavily on the architecture of your system.
+
+   a. You may have some functionality in your library that's incompatible with .NET Core because it relies too heavily on .NET Framework or Windows-specific functionality.  Is it worth leaving that functionality behind for now and releasing a .NET Core verson of your library with less features for the time being?
+
+3. Are there alternative .NET Core APIs for incompatible .NET Framework APIs you're using?
+
+   a. For example, if you're using `WebClient` you may be able to use `HttpClient` instead.
+   
+   b. You may also consider using a slightly different .NET Core API to do some work even if it's not a drop-in replacement.  For example, if you don't require XML Schema discovery, you could use Linq to Xml to parse XML data instead of using XMLSchema to handle the parsing.
+   
+4. Is it reasonable to write some code a .NET Framework API was doing yourself?
+
+   a. Consider modifying and using code from the [.NET Framework Reference Source](referencesource.microsoft.com).  It's licensed under the [MIT License](https://github.com/Microsoft/referencesource/blob/master/LICENSE.txt) so you have significant freedom in doing this.
+   
```

Comment:
Don't like referencesource.microsoft.com because not all code is licensed as MIT. **Only** link to the GitHub repo.